cloudigrade/account/reports/__init__.py

```diff
@@ -179,3 +187,113 @@ def _calculate_instance_usage(start, end, events):
         time_running += diff.total_seconds()
 
     return time_running
+
+def validate_event(event, start):
+    """
+    Ensure that the event is relevant to our time frame.
+
+    Args:
+        event: (InstanceEvent): The event object to evaluate
+        start (datetime.datetime): Start time (inclusive)
+
+    Returns:
+        bool: A boolean regarding whether or not we should inspect the event further.
+    """
+    valid_event = True
+    # if the event occurred outside of our specified period (ie. before start) we should
+    # only inspect it if it was a power on event
+    if event.occurred_at < start:
+        if event.event_type == InstanceEvent.TYPE.power_off:
+            valid_event = False
+    return valid_event
+
+def validate_account_creation(account, end):
+    """
+    Ensure that the event is relevant to our time frame.
+
+    Args:
+        account: (Account): The event object to evaluate
+        end (datetime.datetime): End time (exclusive)
+
+    Returns:
+        bool: A boolean regarding whether or not the account was created after the end
+        time.
+    """
+    if end <= account.created_at:
+        return False
+    return True
+
+
+def get_account_overview(account, start, end):
+    """
+    Generate an overview of an account over a specified amount of time.
+
+    Args:
+        start (datetime.datetime): Start time (inclusive)
+        end (datetime.datetime): End time (exclusive)
+        account (Account or AwsAccount): Account or AwsAccount object
+
+    Returns:
+        dict: An overview of the instances/images/rhel & openshift images for the specified
+        account during the specified time period."""
+    machine_images = AwsMachineImage.objects.all()
```

Comment:
This is redundant and risks introducing performance issues by fetching _all_ images. You could simply remove this line and change your filter below...